### PR TITLE
diffoscope: fix on aarch64-darwin

### PR DESCRIPTION
**Pull request by lf-**
This was comparing stdenv.hostPlatform (an attr set) to a string. This comparison always returned not-equal, which resulted in trying to evaluate the broken gnumeric on macOS. This change fixes that to compare the "system" string as was intended originally.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [x] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lf-** [comment]
Still broken with actual test errors now that I've built it on the community aarch64-darwin machine: https://gist.github.com/lf-/2e7685146861f71795c66f45bd6d6cf7

Nevertheless imo this should be merged. Not sure what to do about the tests. At least one of them looks suspiciously like it is simply broken on darwin due to upstream. My inclination is to just disable them and deal with it later, but shrug.

File: pkgs/tools/misc/diffoscope/default.nix [hunk 1]
```diff
@@ -219,7 +219,7 @@ python3.pkgs.buildPythonApplication rec {
     # oggvideotools is broken on Darwin, please put it back when it will be fixed?
     ++ lib.optionals stdenv.isLinux [ oggvideotools ]
     # This doesn't work on aarch64-darwin
-    ++ lib.optionals (stdenv.hostPlatform != "aarch64-darwin") [ gnumeric ]
+    ++ lib.optionals (stdenv.system != "aarch64-darwin") [ gnumeric ]
```
Comment (RaitoBezarius) [inline pkgs/tools/misc/diffoscope/default.nix:222]:
```suggestion
    ++ lib.optionals (stdenv.hostPlatform.system != "aarch64-darwin") [ gnumeric ]
```

no?

**Comment by lf-** [comment]
I have pushed a commit to Simply Disable The Broken Tests.

Diagnosis:
- `test_libmix_differences` possibly a consequence of us having different bintools
- `test_item_rdb` definitely not our fault, seems to be broken due to rdata version differences or something of the sort
- `test_trim_stderr_in_command` probably caused by /dev/stderr being weird in the sandbox or something like that, definitely straight up our fault.

I have written a bug report to file upstream and will do so when I get a Debian salsa account. *sighs at antispam*

**Comment by lf-** [comment]
Upstream bug here: https://salsa.debian.org/reproducible-builds/diffoscope/-/issues/364

**Pull request merged**